src/main/java/taskat4/Human.java

```diff
@@ -0,0 +1,91 @@
+package taskat4;
+
+import java.io.IOException;
+
+public abstract class Human {
+
+    private final boolean gender;
+    private String name;
+    private String surname;
+    private float height;
+    private float weight;
+
+    public Human(boolean gender, String name, String surname, float height, float weight) {
+        this.gender = gender;
+        this.name = name;
+        this.surname = surname;
+        this.height = height;
+        this.weight = weight;
+    }
+
+    public String toString(boolean bo) {
+        String str;
+        if (bo) str = "Male";
+        else str = "Female";
+        return ("This human " + name + " and surname " + surname + ", is " + height + " cm tall and " + weight + "weight kilo is " + str);
+    }
+
+    public boolean beTogether(float ek1, float ek2) {
+        if (ek1 > ek2) {
+            if (((ek2 / ek1) < 0.9) && (Math.random() < 0.85))
+                return true;
+            else if
+            ((((ek2 / ek1) > 0.9) && (Math.random() < 0.95)))
+                return true;
+        }
+        return false;
+    }
+
+    public Human canDate(Human human1, Human human2) throws IOException {
```

Comment:
same here - this method should take only 1 argument, you should compare object that would call this method to the object passed as a parameter

Reply:
I would not check the implementations, it makes no sense